gauntlet/packages/gauntlet-serum-multisig/src/commands/multisig.ts

```diff
@@ -0,0 +1,140 @@
+  return class Multisig extends SolanaCommand {
+    command: SolanaCommand
+    multisigAddress
+    multisig
+    address
+    program
+    rawTx
+
+    static id = `${command.id}`
+
+    constructor(flags, args) {
+      super(flags, args)
+      logger.info(`Serum multisig wrapping of ${command.id}`)
+
+      this.command = new command(flags, args)
+      this.command.invokeMiddlewares(this.command, this.command.middlewares)
+    }
+
+    createProposal = async (rawTx: RawTransaction) => {
+      logger.loading(`Creating proposal`)
+      const txSize = 1000
+      const transaction = Keypair.generate()
+      const txPublicKey = transaction.publicKey
+      const tx = await this.program.rpc.createTransaction(rawTx.programId, rawTx.accounts, rawTx.data, {
+        accounts: {
+          multisig: this.multisigAddress,
+          transaction: txPublicKey,
+          proposer: this.wallet.payer.publicKey,
+          rent: SYSVAR_RENT_PUBKEY,
+        },
+        instructions: [await this.program.account.transaction.createInstruction(transaction, txSize)],
+        signers: [transaction, this.wallet.payer],
+      })
+      return { tx, txPublicKey }
+    }
+
+    approveProposal = async (txAccount) => {
+      logger.loading(`Approving proposal`)
+      const tx = await this.program.rpc.approve({
+        accounts: {
+          multisig: this.multisigAddress,
+          transaction: txAccount,
+          owner: this.wallet.publicKey,
+        },
+      })
+      return tx
+    }
+
+    executeProposal = async (txAccount) => {
+      logger.loading(`Executing proposal`)
+      this.require(this.flags.tx != null, 'Please set the TX account to execute, using --tx= flag')
+      const [multisigSigner] = await PublicKey.findProgramAddress(
+        [this.multisigAddress.toBuffer()],
+        this.program.programId,
+      )
+      const txAccountData = await this.program.account.transaction.fetch(this.flags.tx)
+      const tx = await this.program.rpc.executeTransaction({
+        accounts: {
+          multisig: this.multisigAddress,
+          multisigSigner,
+          transaction: txAccount,
+        },
+        remainingAccounts: txAccountData.accounts
+          .map((t: any) => {
+            if (t.pubkey.equals(multisigSigner)) {
+              return { ...t, isSigner: false }
+            }
+            return t
+          })
+          .concat({
+            pubkey: txAccountData.programId,
+            isWritable: false,
+            isSigner: false,
+          }),
+      })
+      logger.debug(`TX hash: ${tx.toString()}`)
+
+      return tx
+    }
+
+    execute = async () => {
+      this.multisigAddress = new PublicKey(process.env.MULTISIG_ADDRESS || '')
+      this.multisig = getContract(CONTRACT_LIST.MULTISIG, '')
+      this.address = this.multisig.programId.toString()
+
+      this.program = this.loadProgram(this.multisig.idl, this.address)
+      const txResults = []
+      if (!this.flags.tx) {
```

Comment:
I think we should stick to the 3 steps: create, approve and execute, and perform just one action per execution. 
- Create: Should create the tx proposal, and outputs the the tx public key for future actions.
- Approve: Approves (we can use the `--approve` flag) the proposal created before. We should provide the tx public key with the proposal flag (`--proposal`)
- Executes: Executes the proposal, if the approval is ready

Reply:
Adding the single `--proposal` flag flow as discussed.